api/role.go

```diff
@@ -47,3 +48,66 @@ func (api *API) GetRoleHandler(w http.ResponseWriter, req *http.Request) {
 	log.Event(ctx, "getRole Handler: Successfully retrieved role", log.INFO, logdata)
 
 }
+
+//GetRolesHandler is a handler that gets all roles from MongoDB
+func (api *API) GetRolesHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	//get limit from query parameters, or default value
+	limit, err := getPaginationQueryParameter(req.URL.Query(), "limit", api.defaultLimit)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//get offset from query parameters, or default value
+	offset, err := getPaginationQueryParameter(req.URL.Query(), "offset", api.defaultOffset)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
```

Comment:
This should be "_failed to obtain **offset** from request query parameters_". 
Also is there value in including the error as a param in the log message? i.e
```
log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR, log.Error(err))
```